gms/gossiper.cc

```diff
@@ -842,6 +842,22 @@ future<> gossiper::failure_detector_loop() {
                     return g.failure_detector_loop_for_node(node, generation_number, live_endpoints_version);
                 });
             });
+            std::unordered_set<gms::inet_address> nodes_down;
+            for (const auto& node : nodes) {
+               auto it = std::find(_live_endpoints.begin(), _live_endpoints.end(), node);
```

Comment:
nit: indentation is 3 spaces instead of 4.

Reply:
And it seems like it would be more efficient to keep `nodes` as an unordered_set rather than nodes_down and here walk over `_live_endpoints` and erase the live nodes one by one.

Reply:
Fixed the indentation.

Reply:
I want to avoid making nodes which stores _live_endpoints as unordered_set if possible: big allocation if the number of nodes are big. We need an index to the vector to hash the shard a few lines above, using unordered_set for nodes will not work. Yes if nodes or _live_endpoints are unordered_set it would be a bit faster. However, since this happens on a slow path, I am not sure it is worth the change.